model/src/test/kotlin/DependencyGraphNavigatorTest.kt

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (C) 2021 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.model
+
+import io.kotest.assertions.throwables.shouldThrow
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.shouldNotBe
+
+import org.ossreviewtoolkit.model.config.AnalyzerConfiguration
+import org.ossreviewtoolkit.utils.Environment
+
+class DependencyGraphNavigatorTest : AbstractDependencyNavigatorTest() {
+    override val resultFileName = RESULT_FILE
+
+    override val resultWithIssuesFileName = RESULT_WITH_ISSUES_FILE
+
+    init {
+        "init" should {
+            "fail if there is no analyzer result" {
+                shouldThrow<IllegalArgumentException> {
+                    DependencyGraphNavigator(OrtResult.EMPTY)
+                }
+            }
+
+            "fail if the map with dependency graphs is empty" {
+                val result = OrtResult.EMPTY.copy(
+                    analyzer = AnalyzerRun(
+                        result = AnalyzerResult.EMPTY,
+                        environment = Environment(),
+                        config = AnalyzerConfiguration()
+                    )
+                )
+
+                shouldThrow<IllegalArgumentException> {
+                    DependencyGraphNavigator(result)
+                }
+            }
+        }
+
+        "DependencyNode.equals" should {
```

Comment:
What is the reason not to add these test cases to the abstract class?

Reply:
They have originally been written specifically for the `equals()` implementation of the internal `DependencyRefCursor` class; but you are right, the tests can as well be moved to the base class to check `equals()` for all `DependencyNode` implementations. Done.